scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/x509/KeyManagerCertificateProvider.java

```diff
@@ -363,6 +363,12 @@ private boolean matchCurves(List<SupportedGroup> curves, List<X509Certificate> c
 		for (X509Certificate certificate : chain) {
 			PublicKey certPublicKey = certificate.getPublicKey();
 			if (Asn1DerDecoder.isSupported(certPublicKey.getAlgorithm())) {
+			    // for rsa key we only check whether server support certain curves for key exchange.
+				// As curves is already negotiated based on server config so we always return true
```

Comment:
I have my doubts about the statement of that comment. Sure the common curves have been negotiated at that point, but if you consider "mixed certificate chains" (RSA+EC), then a check for the other (EC) Certificates will be required. With that, I would change the `return` logically in `continue`. I have seen some other places, where something as "isEC" may improve to distinguish between EC based stuff (requires curve check), and non EC based stuff. In the past [SignatureAlgorithm.isEcdsaCompatible()](https://github.com/eclipse/californium/blob/master/scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/SignatureAndHashAlgorithm.java#L233) was useful, but maybe a more general solution is required.

Reply:
You are right, I should change return to contine. I failed to consider the "mixed certificate chains" situation.

Reply:
I will fix this when I got home. (I submitted this code from my personal computer, uploading code from company computer requires too many paperwork I simply gave up....). 
Should I amend this commit and perform I force push or just add a normal commit?